lib/util/pssh.js

```diff
@@ -189,4 +208,43 @@ shaka.util.Pssh = class {
 
     return shaka.util.Uint8ArrayUtils.concat(...dedupedInitDatas);
   }
+
+  /**
+   * Get a pssh from the given keySystem and moov.
+   *
+   * @param {!Uint8Array} initData
+   * @param {string} keySystem
+   * @return {!Uint8Array}
+   */
+  static getPsshFromInitData(initData, keySystem) {
+    const systemId = shaka.util.Pssh.defaultSystemIds_.get(keySystem);
+    const psshInfo = new shaka.util.Pssh(initData, true);
+
+    // If there is only a single pssh, return the original array.
+    if (psshInfo.dataBoundaries.length <= 1) {
+      return initData;
+    }
+
+    const index = psshInfo.systemIds.indexOf(systemId);
+    // If the index is -1 we can not found the correct pssh
+    if (index === -1) {
+      return initData;
+    }
+
+    // 8 due to moov header
```

Comment:
This seems a little hacky.  First, you assume it's coming from inside moov, which the mp4 parser doesn't guarantee.  Second, you assume that the pssh box is the first child of the moov, which may not be true.

Perhaps you can make the Pssh parser aware of the offset at which its own box starts, such that dataBoundaries contains the correct offsets.

Reply:
Ok, I changed it. I added a boxOffset to parseBox_ and I initialized correctly for each case.